src/lib/classes/notification.js

```diff
@@ -0,0 +1,512 @@
+      const schema = await result.json();
+
+      this.schema = {
+        ...this.schema,
+        [name]: schema
+      };
+    } catch (error) {
+      throw new SolidError(error.message, 'Fetch Shape', error.status);
+    }
+  };
+
+  buildShapeObject = shape => {
+    return {
+      name: (shape && shape.name) || this.shapeName,
+      shape: (shape && shape.path) || this.shape
+    };
+  };
+
+  /**
+   * create and send a notification to user's inbox
+   * @param inboxRoot
+   * @param title
+   * @param content
+   * @param options
+   * @returns {Promise<*>}
+   */
+
+  create = async (content = {}, to, options = {}) => {
+    try {
+      const currentShape = this.buildShapeObject(options && options.shape);
+      const { name, shape: defaultShape } = currentShape;
+      const termFactory = N3.DataFactory;
+      const { namedNode, literal } = termFactory;
+
+      if (!this.schema || (this.schema && !this.schema[name])) {
+        /**
+         * Fetch remote shape that will be use to render and build notifications
+         */
+        await this.fetchNotificationShape(defaultShape, name);
+      }
+      if (!this.schema[name]['@context']) {
+        throw new SolidError('Schema does not have context', 'Notification', 500);
+      }
+
+      const { '@context': context, shape } = this.schema[name];
+
+      /**
+       * N3 is an implementation of the RDF.js low-level specification that lets you handle RDF in JavaScript easily.
+       * Reference: https://www.npmjs.com/package/n3
+       * Create content for the notification using turtle format.
+       * @type {N3Writer}
+       */
+      const writer = new N3.Writer({
+        prefixes: context,
+        format: 'text/turtle'
+      });
+
+      shape.forEach(item => {
+        if (item.property && item.property.includes(':')) {
+          const defaultValue = item.value;
+
+          if (
+            content[item.label] ||
+            defaultValue ||
+            item.label === 'read' ||
+            item.label === 'datetime'
+          ) {
+            /**
+             * Add read by default on notification document
+             * @type {string}
+             */
+            let value = item.label === 'read' ? 'false' : content[item.label];
+            /**
+             * Add datetime time by default on notification document
+             * @type {string}
+             */
+            value = item.label === 'datetime' ? new Date().toISOString() : value;
+            /**
+             * Check if object from schema is a literal or node value
+             */
+            const typedValue = item.type === 'NamedNode' ? namedNode(value) : literal(value);
+
+            writer.addQuad(
+              namedNode(''),
+              namedNode(`${context[item.property.split(':')[0]]}${item.label}`),
+              typedValue
+            );
+          }
+        } else {
+          throw new SolidError('Schema do not have property', 'Notification', 500);
```

Comment:
This sounds clunky, should it be Schema does not have property? Or is this multiple schema (plural)?

Reply:
Updated to: Schema does not have property